src/test/java/io/inscopemetrics/kairosdb/aggregators/AbstractHistogramTest.java

```diff
@@ -41,56 +41,72 @@
+            Arrays.asList(
+                    CREATE_HISTOGRAM_V1_FROM_COUNTS
+            ),
+            Arrays.asList(
+                    CREATE_HISTOGRAM_V2_FROM_COUNTS
+            ));
+
     /**
-     * Create a parameterization per histogram from values factory of the
-     * supplied parameters.
+     * Create a parameterization per histogram type providing both from values
+     * and from counts factories.
      *
-     * @param parametersPerFactory The parameters to include with each factory.
      * @return The resulting parameterization by factory.
      */
-    public static Collection<Object[]> createParametersForHistogramFromValues(
-            final Collection<Object> parametersPerFactory) {
-        final List<Object[]> parameterSets = new ArrayList<>();
-        for (final CreateHistogramFromValues factory : getHistogramFactoriesFromValues()) {
-            final Object[] parameters = new Object[parametersPerFactory.size() + 1];
-
-            // The first parameter is the factory
-            parameters[0] = factory;
-
-            // The next parameters are the ones supplied
-            int i = 1;
-            for (final Object parameter : parametersPerFactory) {
-                parameters[i] = parameter;
-                ++i;
-            }
+    public static Collection<Object[]> createParameters() {
+        return createParameterization(ALL_PARAMETERS_BY_TYPE);
+    }
 
-            // Add the parameter set
-            parameterSets.add(parameters);
-        }
-        return parameterSets;
+    /**
+     * Create a parameterization per histogram type providing only from values
+     * factories.
+     *
+     * @return The resulting parameterization by factory.
+     */
+    public static Collection<Object[]> createParametersFromValues() {
+        return createParameterization(FROM_VALUES_PARAMETERS_BY_TYPE);
     }
 
     /**
-     * Create a parameterization per histogram from counts factory of the
-     * supplied parameters.
+     * Create a parameterization per histogram type providing only from counts
+     * factories.
      *
-     * @param parametersPerFactory The parameters to include with each factory.
      * @return The resulting parameterization by factory.
      */
-    public static Collection<Object[]> createParametersForHistogramFromCounts(
-            final Collection<Object> parametersPerFactory) {
-        final List<Object[]> parameterSets = new ArrayList<>();
-        for (final CreateHistogramFromCounts factory : getHistogramFactoriesFromCounts()) {
-            final Object[] parameters = new Object[parametersPerFactory.size() + 1];
+    public static Collection<Object[]> createParametersFromCounts() {
+        return createParameterization(FROM_COUNTS_PARAMETERS_BY_TYPE);
+    }
 
-            // The first parameter is the factory
-            parameters[0] = factory;
+    private static Collection<Object[]> createParameterization(
+            final List<List<Object>> parameterListA
```

Comment:
`parameterLists`?

I'm not sure what the A refers to, and there isn't a B

Reply:
See above.